daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/FrontStack.scala

```diff
@@ -151,7 +150,11 @@ object FrontStack extends FrontStackInstances {
   def empty[A]: FrontStack[A] = emptySingleton
 
   def apply[A](xs: ImmArray[A]): FrontStack[A] =
-    new FrontStack(FQPrepend(xs, FQEmpty), length = xs.length)
+    if (xs.length > 0) {
+      new FrontStack(FQPrepend(xs, FQEmpty), length = xs.length)
+    } else {
+      empty
```

Comment:
Can you add a test case for this?

Reply:
I don't see how I could test this in the given test structure. `FQPrepend` is a private implementation detail that seems to be unreachable from the tests.

Reply:
Fair. 🙂